song_rater/tests.py

```diff
@@ -23,79 +45,109 @@ def test_add_song_form(self):
-        self.test_song_list_one_song()
+        test_values_1 = {
+            'artist': 'Strawberry',
+            'title': 'Shortcake',
+            'rating': 4
+        }
+        test_values_2 = {
+            'artist': 'Lemon',
+            'title': 'Meringue',
+            'rating': 3
+        }
+        add_song_and_rating(**test_values_1)
+        add_song_and_rating(**test_values_2)
+
+        response = self.client.get('/song-list')
+
+        for value in {*test_values_1.values(), *test_values_2.values()}:
+            with self.subTest():
+                self.assertContains(response, value)
 
 
 class SongDetailTestCase(TestCase):
     def test_rating_list_empty(self):
-        song = add_song(artist='Apple', title='Pie')
-        response = self.client.get(f'/{song.pk}', follow=True)
+        test_song = {
+            'artist': 'Apple',
+            'title': 'Pie',
+        }
+        test_song = add_song(**test_song)
+        response = self.client.get(f'/{test_song.pk}', follow=True)
         self.assertContains(response, 'No ratings available yet. Try adding a new rating')
 
     def test_rating_list_one_rating(self):
-        rating = 4
-        song = add_song_and_rating(artist='Blueberry', title='Muffin', rating=rating)
-        response = self.client.get(f'/{song.pk}', follow=True)
-        self.assertContains(response, song.artist, count=1)
-        self.assertContains(response, song.title, count=1)
-        self.assertContains(response, rating)
+        test_values = {
+            'artist': 'Blueberry',
+            'title': 'Muffin',
+            'rating': 4,
+        }
+        test_song = add_song_and_rating(**test_values)
+        response = self.client.get(f'/{test_song.pk}', follow=True)
+
+        for value in list(test_values.values()):  # loop over all test values
+            with self.subTest():
+                self.assertContains(response, value)
 
     def test_rating_list_multiple_ratings(self):
-        song = add_song(artist='Lemon', title='Meringue')
+        test_song = {
+            'artist': 'Lemon',
+            'title': 'Meringue',
+        }
+        test_song = add_song(**test_song)
         ratings = np.random.randint(low=1, high=6, size=10)
 
         for rating in ratings:
-            add_rating(song=song, rating=rating)
+            add_rating(song=test_song, rating=rating)
 
-        response = self.client.get(f'/{song.pk}', follow=True)
-        self.assertContains(response, song.artist, count=1)
-        self.assertContains(response, song.title, count=1)
+        response = self.client.get(f'/{test_song.pk}', follow=True)
+        self.assertContains(response, test_song.artist, count=1)
+        self.assertContains(response, test_song.title, count=1)
 
         for rating in ratings:
-            self.assertContains(response, rating)
+            with self.subTest():
+                self.assertContains(response, rating)
```

Comment:
Je kunt hier ook nog controleren of het gemiddelde van de ratings terugkomt in de lijst. 

Het gemiddelde is `sum(ratings)/10`

Reply:
Nice find! I added it to the test.